src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,179 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import command.Command;
+import command.DeadlineCommand;
+import command.DeleteCommand;
+import command.EventCommand;
+import command.FindCommand;
+import command.ListCommand;
+import command.MarkCommand;
+import command.TodoCommand;
+import command.UnmarkCommand;
+
+/**
+ * Parses user input
+ */
+public class Parser {
+    private static final DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
```

Comment:
Good use of a constant to avoid magic strings/numbers in code to enhance readability and maintainability.